### ISSUE TITLE
Using Every Door in fully offline mode (no pre-caching of OSM tiles?)

### ISSUE THREAD

**Issue by mnalis** [Zverik/every_door#589]
I've recently been in position where I needed to map an area, without having Internet access most of the time (I have it available only once a day). I did try browsing over all area that I'd be visiting to preload data, as well using download button to get all OSM data in the area. I also downloaded satellite imagery in its display download button.

However, the result was disappointing. The OSM map tiles were only visible on one zoom level, and invisible (black screen) on all others. That made mapping new POIs (which was primary the case) extremely hard to impossible.

Possible ideas to make EveryDoor more offline-friendly in this respect (more than one can be implemented for best effect!):

- like for satellite imagery, add a download button to download regular OSM (Carto?) tiles in all zoom levels, so interested parties can pre-download the area where there is no internet coverage. Probably the easiest to implement (as the code for downloading for other TMS layers already exist)?
- if exact zoom is unavailable and cannot be downloaded, resize zoom level that is available to fit (that will distort the image, but will still be better than blank screen one gets currently)
- use vector-based tiles (that would be ideal as downloaded detailed vectors would work at zooms without distortions, and use less data traffic and storage space as additional benefit, but I don't know if the rendering libraries used by EveryDoor support that)

**Comment by Zverik** [comment]
I know this is not easily discoverable, but on the satellite layer page there's a small button at the top right corner. It downloads all tiles for downloaded data areas :)

**Comment by mnalis** [comment]
> I know this is not easily discoverable, but on the satellite layer page there's a small button at the top right corner. It downloads all tiles for downloaded data areas :)

Yes, it it well hidden, but I'm aware of it :smile:. That's what I meant by _"I also downloaded satellite imagery in its display download button."_ above.

But I think it pre-downloads **only** satellite TMS, *not* OSM Carto TMS. If it is supposed to **also** pre-download OSM Carto TMS, it does not seem to work (in 3.0 at least, still waiting for F-droid to publish 3.1)

Here is example what I mean:

https://github.com/Zverik/every_door/assets/156656/c02dfdb1-1008-45ac-8776-936489c3cf7a


Even when I clicked on that button to pre-download, OSM Carto TMS is not pre-downloaded (at least not at bigger zoom levels, as you can see the screen turn blank when I zoom in at 15s-23s in the video above).

What I would like, is that OSM Carto TMS (in all zoom levels for that area) is also preloaded when that button in pressed, so it can be used offline.

**Comment by Zverik** [comment]
OSM tiles are downloaded, as you can see [here](https://github.com/Zverik/every_door/blob/main/lib/providers/tile_cacher.dart#L114), but the max zoom level is limited to 18, which might be the reason you're getting grey tiles in the video. You are zooming to level 19.

**Comment by pandadeepimpact** [comment]
> The OSM map tiles were only visible on one zoom level, and invisible (empty screen) on all others. That made mapping new POIs (which was primary the case) extremely hard to impossible.

I, too, frequently encounter this problem as I use the offline mode almost exclusively when mapping. The maximum zoom level that Every Door has is too small for a city, where an area that size usually contain 8 stores or more, a huge list that is more confusing than simply zooming in to see what store I'm standing next to. SCEE suffers no such problem.

I really hope that this gets fixed as it breaks one of the main goals of Every Door, speed, especially in areas predominated by small stores or kiosks.

**Comment by mnalis** [comment]
If download for all zoom levels is prohibitive, perhaps Every Door can at least try to **extrapolate missing zoom levels**? 

E.g. if the data is downloaded up to zoom `18`, and I zoom to `19`, if the Internet connection is missing, I should **not** get a **blank screen**, but data from zoom `18` should be upscaled and displayed instead (with some loss of quality, of course, but still very much more usable than blank screen!)
